### PR TITLE
Rename Navigation{Mesh,Polygon}Instance and PlaneShape for clarity

### PR DESCRIPTION
**Pull request by AndreaCatania**
Renamed:
- `NavigationMeshInstance` -> `NavigationRegion`
- `PlaneShape` -> `WorldMarginShape`
- `NavigationPolygonInstance` -> `Navigation2DRegion`

### CODE REVIEW

**Comment by Calinou** [comment]
> PlaneShape -> WorldMarginShape

This one is probably less confusing, but what about renaming it to InfinitePlaneShape so that existing users can find the new class more easily?

**Comment by aaronfranke** [comment]
Why not include the `3D` suffix? Would be strange to rename them here and then again in #30736

**Comment by AndreaCatania** [comment]
@Calinou the `InfinitePlane` doesn't solve the main problem because user will continue to use it as ground. For me `InfinitePlaneShape` is nice but I feel `WorldMarginShape` to be much more clear why that shape exist.

@aaronfranke yes, but I prefer that `3D` is added all together with the PR proposal so that change is uniform.

**Comment by AndreaCatania** [comment]
Also renamed `NavigationPolygonInstance` to `Navigation2DRegion`

**Comment by akien-mga** [comment]
> Also renamed `NavigationPolygonInstance` to `Navigation2DRegion`

Shouldn't it be `NavigationRegion2D`? Most 2D nodes have 2D as suffix, and we'll eventually add 3D suffixes to 3D nodes, so we'd also have `NavigationRegion3D`.

**Comment by AndreaCatania** [comment]
Yes, correct - I followed the server name convention, I'll rename it.

**Comment by AndreaCatania** [comment]
Name fixed, @akien-mga .

File: scene/3d/navigation_region.h [hunk 1]
```diff
@@ -28,18 +28,18 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
 
-#ifndef NAVIGATION_MESH_INSTANCE_H
-#define NAVIGATION_MESH_INSTANCE_H
+#ifndef navigation_region_H
```
Comment (akien-mga) [inline scene/3d/navigation_region.h]: I'd prefer to keep the guards uppercase.

**Comment by akien-mga** [comment]
Can you add the old names as compatibility aliases in `scene/register_scene_types.cpp`? Grep for `ClassDB::add_compatibility_class`.

File: scene/3d/navigation_region.h (hunk 1 above)
Comment (AndreaCatania) [reply scene/3d/navigation_region.h]: Done

File: scene/resources/world_margin_shape.h [hunk 2]
```diff
@@ -28,14 +28,14 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
 
-#ifndef PLANE_SHAPE_H
-#define PLANE_SHAPE_H
+#ifndef world_margin_shape_H
```
Comment (akien-mga) [inline scene/resources/world_margin_shape.h]: Still lowercase here.

**Review by akien-mga — APPROVED** [review]

**Comment by AndreaCatania** [comment]
All right! fixed

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**